cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1161,7 +1180,7 @@ SpecificAndLatestVersionKeys LocalVersionedEngine::get_stream_index_map(
 }
 
 folly::Future<std::pair<std::optional<VariantKey>, std::optional<google::protobuf::Any>>> LocalVersionedEngine::get_metadata(
-    std::optional<AtomKey> key){
+    std::optional<AtomKey>&& key){
```

Comment:
They have different meanings. In a non-template context, && will require an r-value, whereas the plain version will copy then move

Reply:
The team should probably settle on a style for the general case (i.e. beyond the move c'tor/assignment which has to use `&&`).

I avoid `&&` on arguments that will be moved internally in a function:

Pros: 
* give the caller a clear choice to either copy or `move` when passing this parameter
* easier to understand and reason (the object's lifetime clearly belongs to this function)

Cons:
* the `move` at the call site might not be optimised out
* can easily cause a copy by mistake

Reply:
In this case, I want to explicitly indicate I want an r-value, to avoid unnecessary performance mistakes. I think that adding && is also an indicator of what we are trying to do. In this case, I do want to move the key because the returned metadata already contains the key as well (DecodeMetadataTask ends up returning  std::pair<std::optional<VariantKey>, std::optional<google::protobuf::Any>>.

Reply:
but yes, it is a good idea IMO to settle some of the members of our team and reach some agreement about common practices